draft-ietf-lamps-dilithium-certificates.md

```diff
@@ -394,18 +394,41 @@ in this section.
-
-Note that the signing mode defined here is different from HashML-DSA defined in [FIPS204] section 5.4. This specification uses exclusively ExternalMu-ML-DSA for pre-hashed use cases, and thus HashML-DSA as defined in [FIPS204] and identified by `id-hash-ml-dsa-44-with-sha512`, `id-hash-ml-dsa-65-with-sha512`, and `id-hash-ml-dsa-87-with-sha512` MUST NOT be used in X.509 and related PKIX protocols.
-
-All functions and notation used in {{fig-externalmu-ml-dsa-external}} and {{fig-externalmu-ml-dsa-internal}} are defined in [FIPS204].
+# Pre-hashing (ExternalMu-ML-DSA) {#prehash}
+
+Some applications require prehashing, where the signature generation
+process can be separated into a pre-hash step and a core signature
+step in order to ease operational requirements around large or
+inconsistently-sized payloads. This can be performed at the
+protocol layer, but not all protocols support it.
+An example in [RFC5280] is the certificate revocation list
+(CRL) data structures, that do not include message digesting before signing.
+This makes signing large volumes of large CRLs challenging.
+
+As mentioned in the introduction, pure ML-DSA signing itself
+supports a prehashing flow by splitting the operation over two
+modules. In this section we make this "ExternalMu-ML-DSA"
+more explicit.
+
+There are two steps. First an `ExternalMu-ML-DSA.Prehash()`
+followed by `ExternalMu-ML-DSA.Sign()`. Together these operate
+as `ML-DSA.Sign()` from [FIPS204] and thus create exactly the same
+signatures as regular pure ML-DSA.
+An ML-DSA key and certificate MAY be used with either ML-DSA
+or ExternalMu-ML-DSA interchangeably.
+Note that ExternalMu-ML-DSA describes a different signature API from ML-DSA
+and therefore might require explicit support from hardware or
+software cryptographic modules.
+
+Note that the signing mode defined here is different from HashML-DSA
+defined in [FIPS204] section 5.4. This specification uses exclusively
+ExternalMu-ML-DSA for pre-hashed use cases, and thus HashML-DSA as
+defined in [FIPS204] and identified by `id-hash-ml-dsa-44-with-sha512`,
+`id-hash-ml-dsa-65-with-sha512`, and `id-hash-ml-dsa-87-with-sha512`
+MUST NOT be used in X.509 and related PKIX protocols.
```

Comment:
Suggest:

... public keys identified by these OIDs MUST NOT be used within X.509 certificates, and the signature algorithms identified by these OIDs MUST NOT be used within X.509 and related PKIX protocols.

Reply:
Will tweak in separate PR.